engine/api/ascode/ascode_pr.go

```diff
@@ -0,0 +1,167 @@
+package ascode
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/cache"
+	"github.com/ovh/cds/engine/api/event"
+	"github.com/ovh/cds/engine/api/operation"
+	"github.com/ovh/cds/engine/api/repositoriesmanager"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+const (
+	AsCodePipeline = "pipeline"
+	AsCodeWorkflow = "workflow"
+)
+
+type EntityData struct {
+	Type      string
+	ID        int64
+	Name      string
+	FromRepo  string
+	Operation *sdk.Operation
+}
+
+// UpdateAsCodeResult pulls repositories operation and the create pullrequest + update workflow
+func UpdateAsCodeResult(ctx context.Context, db *gorp.DbMap, store cache.Store, p *sdk.Project, app *sdk.Application, ed EntityData, u sdk.Identifiable) {
```

Comment:
We should have a unit tests with mocks on this function

Reply:
done in workflow_run_test.go